titus-ext/eureka/src/test/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthServiceTest.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    @Test
+    public void testJobManagerUpdate() {
+        StepVerifier.create(healthService.events(false))
+                // Task launched, but not in Eureka yet.
+                .then(() -> jobManagerStub.moveTaskToState(taskId1, TaskState.Launched))
+                .assertNext(event -> {
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Unknown);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Unknown);
+                })
+
+                // Task started and registered with Eureka
+                .then(() -> {
+                    eurekaServer.register(newInstanceInfo(taskId1, InstanceStatus.UP));
+                    jobManagerStub.moveTaskToState(taskId1, TaskState.Started);
+                })
+                .assertNext(event -> {
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Healthy);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Healthy);
+                })
+
+                // Task terminated
+                .then(() -> jobManagerStub.moveTaskToState(task1, TaskState.Finished))
+                .assertNext(event -> {
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Terminated);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Terminated);
+                })
+
+                .thenCancel()
+                .verify(Duration.ofSeconds(5));
+    }
+
+    @Test
+    public void testEurekaUpdate() {
+        jobManagerStub.moveTaskToState(taskId1, TaskState.Started);
+
+        StepVerifier.create(healthService.events(false))
+                // Change state to UP
+                .then(() -> {
+                    eurekaServer.register(newInstanceInfo(taskId1, InstanceStatus.UP));
+                    eurekaServer.triggerCacheRefreshUpdate();
+                })
+                .assertNext(event -> {
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Healthy);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Healthy);
+                })
+                // Change state to DOWN
+                .then(() -> {
+                    eurekaServer.register(newInstanceInfo(taskId1, InstanceStatus.DOWN));
+                    eurekaServer.triggerCacheRefreshUpdate();
+                })
+                .assertNext(event -> {
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Unhealthy);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Unhealthy);
+                })
+
+                .thenCancel()
+                .verify(Duration.ofSeconds(5));
+    }
+
+    @Test
+    public void testEurekaStaleDataCleanup() {
```

Comment:
Add test for when eureka status goes from `UP` to `NOT_REGISTERED`

Reply:
Done